roles/crawler/defaults/main.yml

```diff
@@ -1,20 +1,91 @@
 ---
-crawler_apt_packages:
-  - xvfb # tbs
+# Whether to configure the project application code.
+# Allows skipping the code tasks if only the database
+# tasks should be run, which is useful in a multi-host setup.
+fpsd_crawler_configure_service: true
+
+# General system dependencies requiring for running
+# Tor Browser Bundle (tbb).
+fpsd_crawler_apt_packages:
+  - aptitude # needed by Ansible for apt 'upgrade'
   - git #tbs
   - gnupg-curl # needed for https keyservers
-  - libfontconfig1 # tbb (firefox)
-  - libxrender1 # tbb (firefox)
   - libasound2 # tbb (firefox)
   - libdbus-glib-1-2 # tbb (firefox)
+  - libfontconfig1 # tbb (firefox)
   - libgtk2.0-0 # tbb (firefox)
+  - libpq-dev # needed to install psycopg2
+  - libxrender1 # tbb (firefox)
+  - python3-pip
+  - python3.5
+  - tmux
+  - vim
+  - xvfb # tbs
+
+# General Python tooling for development. Specific Python dependencies
+# for the crawler code are installed from `requirements.txt` files.
+fpsd_crawler_pypi_packages:
+  - pip
+  - pytest
+  - ipython
+  - ipdb
 
 # Specify TBB version to download and install
-tbb_release: 6.0.5
-tbb_locale: en-US
-tbb_arch: 64
+fpsd_crawler_tbb_release: 6.0.5
+fpsd_crawler_tbb_locale: en-US
+fpsd_crawler_tbb_arch: 64
+fpsd_crawler_tbb_directory: "~{{ fpsd_crawler_system_account }}/tbb"
 
 # Specify Tor version to download, patch, build, and install
-tor_release: 0.2.8.7
+fpsd_crawler_tor_release: 0.2.8.7
+fpsd_crawler_tor_directory: "~{{ fpsd_crawler_system_account }}"
```

Comment:
Writing to `/opt` would need to be done with elevated privileges, and actually `/usr/local/src` would be a more appropriate location here for a system-wide dir. Using the home dir, however, is designed to provide an unprivileged process for downloading, configuring, and compiling the source code; only the install command should involved privilege escalation.

> We don't want to have to build and/or download TBB for multiple users.

The approach of writing to the home dir does not require recompilation; even in multi-user setups, it will only build tor once.